From 6df92e7950b8d29436c7a812c977d65591238f98 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Tue, 3 Jul 2018 10:46:58 +0200
Subject: [PATCH] VisualItemModel: fix updateIndices

UpdateIndices set the indices starting from the unitialized n, when such an
invalid index is used later on, a segfault typically occurs. So set n to 0.
---
 src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp b/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp
index fefbc74..1498f81 100644
--- a/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp
+++ b/src/declarative/graphicsitems/qdeclarativevisualitemmodel.cpp
@@ -104,7 +104,7 @@ public:
     void updateIndices()
     {
         QList<Item>::iterator i;
-        int n;
+        int n = 0;
 
         for (i = children.begin(); i != children.end(); ++i) {
             QDeclarativeVisualItemModelAttached *attached = QDeclarativeVisualItemModelAttached::properties((*i).item);
-- 
2.7.4

